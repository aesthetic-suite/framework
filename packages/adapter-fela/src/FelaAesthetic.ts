import Aesthetic, { AestheticOptions, ClassName, Ruleset, Sheet } from 'aesthetic';
import { combineRules, IRenderer } from 'fela';
import { render } from 'fela-dom';
import { NativeBlock, ParsedBlock } from './types';

export default class FelaAesthetic<Theme extends object> extends Aesthetic<
  Theme,
  NativeBlock,
  ParsedBlock
> {
  fela: IRenderer;

  keyframes: { [animationName: string]: ClassName } = {};

  constructor(fela: IRenderer, options: Partial<AestheticOptions> = {}) {
    super(options);

    this.fela = fela;

    this.syntax
      .on('attribute', this.handleNested)
      .on('css', this.handleCss)
      .on('fallback', this.handleFallback)
      .on('font-face', this.handleFontFace)
      .on('global', this.handleGlobal)
      .on('keyframe', this.handleKeyframe)
      .on('media', this.handleMedia)
      .on('property', this.handleProperty)
      .on('pseudo', this.handleNested)
      .on('selector', this.handleNested)
      .on('support', this.handleSupport);
  }

  flushStyles() {
    render(this.fela);
  }

  transformToClassName(styles: ParsedBlock[]): ClassName {
    return this.fela.renderRule(combineRules(...styles.map(style => () => style)), {});
  }

  purgeStyles() {
    const elements = document.querySelectorAll('style[data-fela-type]');

    if (elements) {
      Array.from(elements).forEach(element => {
        // eslint-disable-next-line no-param-reassign
        element.textContent = '';
      });
    }
  }

  // http://fela.js.org/docs/api/fela/Renderer.html
  private handleCss = (css: string) => {
    this.fela.renderStatic(css);
  };

  // https://github.com/rofrischmann/fela/tree/master/packages/fela-plugin-fallback-value
  private handleFallback = (
    ruleset: Ruleset<NativeBlock>,
    name: keyof NativeBlock,
    value: any[],
  ) => {
    let fallbacks: any = ruleset.properties[name] || [];

    if (!Array.isArray(fallbacks)) {
      fallbacks = [fallbacks];
    }

    ruleset.addProperty(name, [...value, ...fallbacks]);
  };

  // http://fela.js.org/docs/basics/Fonts.html
  // http://fela.js.org/docs/api/fela/Renderer.html
  private handleFontFace = (
    sheet: Sheet<NativeBlock>,
    fontFaces: Ruleset<NativeBlock>[],
    fontFamily: string,
    srcPaths: string[][],
  ) => {
    fontFaces.forEach((face, i) => {
      const { local, ...style } = face.toObject() as any;

      this.fela.renderFont(fontFamily, srcPaths[i], {
        ...style,
        localAlias: local,
      });
    });
  };

  // http://fela.js.org/docs/advanced/StaticStyle.html
  // http://fela.js.org/docs/api/fela/Renderer.html#renderstaticstyle-selector
  private handleGlobal = (
    sheet: Sheet<NativeBlock>,
    selector: string,
    ruleset: Ruleset<NativeBlock>,
  ) => {
    this.fela.renderStatic(ruleset.toObject(), selector);
  };

  // http://fela.js.org/docs/basics/Keyframes.html
  // http://fela.js.org/docs/basics/Renderer.html#renderkeyframe
  private handleKeyframe = (
    sheet: Sheet<NativeBlock>,
    keyframe: Ruleset<NativeBlock>,
    animationName: string,
  ) => {
    this.keyframes[animationName] = this.fela.renderKeyframe(() => keyframe.toObject() as any, {});
  };

  // http://fela.js.org/docs/basics/Rules.html#3-media-queries
  private handleMedia = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@media ${query}`, value);
  };

  // http://fela.js.org/docs/basics/Rules.html#style-object
  private handleNested = (
    ruleset: Ruleset<NativeBlock>,
    selector: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(selector, value);
  };

  // http://fela.js.org/docs/basics/Rules.html
  private handleProperty = (ruleset: Ruleset<NativeBlock>, name: keyof NativeBlock, value: any) => {
    if (name === 'animationName') {
      ruleset.addProperty(name, this.syntax.injectKeyframes(value, this.keyframes).join(', '));
    } else {
      ruleset.addProperty(name, value);
    }
  };

  // http://fela.js.org/docs/basics/Rules.html
  private handleSupport = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@supports ${query}`, value);
  };
}
