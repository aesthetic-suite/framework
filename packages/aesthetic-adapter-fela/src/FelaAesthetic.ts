/**
 * @copyright   2017, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import Aesthetic, { AestheticOptions, ClassName, Keyframes, Ruleset, Sheet } from 'aesthetic';
import { createRenderer, combineRules, IRenderer } from 'fela';
import { render } from 'fela-dom';
import { NativeBlock, ParsedBlock } from './types';

export default class FelaAesthetic<Theme> extends Aesthetic<Theme, NativeBlock, ParsedBlock> {
  fela: IRenderer;

  keyframes: { [animationName: string]: ClassName } = {};

  constructor(fela?: IRenderer, options: Partial<AestheticOptions> = {}) {
    super(options);

    this.fela = fela || createRenderer();

    this.syntax
      .on('attribute', this.handleNested)
      .on('fallback', this.handleFallback)
      .on('font-face', this.handleFontFace)
      .on('global', this.handleGlobal)
      .on('keyframes', this.handleKeyframes)
      .on('media', this.handleMedia)
      .on('property', this.handleProperty)
      .on('pseudo', this.handleNested)
      .on('selector', this.handleNested)
      .on('supports', this.handleSupports);

    render(this.fela);
  }

  // https://github.com/rofrischmann/fela/tree/master/packages/fela-plugin-fallback-value
  handleFallback = (ruleset: Ruleset<NativeBlock>, name: keyof NativeBlock, value: any[]) => {
    let fallbacks: any = ruleset.properties[name] || [];

    if (!Array.isArray(fallbacks)) {
      fallbacks = [fallbacks];
    }

    ruleset.addProperty(name, [...fallbacks, ...value]);
  };

  // http://fela.js.org/docs/basics/Fonts.html
  // http://fela.js.org/docs/api/fela/Renderer.html
  handleFontFace = (
    sheet: Sheet<NativeBlock>,
    fontFaces: NativeBlock[],
    fontFamily: string,
    srcPaths: string[][],
  ) => {
    fontFaces.map((face, i) => {
      const { local, ...style } = face as any;

      this.fela.renderFont(fontFamily, srcPaths[i], {
        ...style,
        localAlias: local,
      });
    });
  };

  // http://fela.js.org/docs/advanced/StaticStyle.html
  // http://fela.js.org/docs/api/fela/Renderer.html#renderstaticstyle-selector
  handleGlobal = (sheet: Sheet<NativeBlock>, selector: string, ruleset: Ruleset<NativeBlock>) => {
    this.fela.renderStatic(ruleset.toObject(), selector);
  };

  // http://fela.js.org/docs/basics/Keyframes.html
  // http://fela.js.org/docs/basics/Renderer.html#renderkeyframe
  handleKeyframes = (
    sheet: Sheet<NativeBlock>,
    keyframes: Keyframes<NativeBlock>,
    animationName: string,
  ) => {
    this.keyframes[animationName] = this.fela.renderKeyframe(() => keyframes as any, {});
  };

  // http://fela.js.org/docs/basics/Rules.html#3-media-queries
  handleMedia = (ruleset: Ruleset<NativeBlock>, query: string, value: Ruleset<NativeBlock>) => {
    ruleset.addNested(`@media ${query}`, value);
  };

  // http://fela.js.org/docs/basics/Rules.html#style-object
  handleNested = (ruleset: Ruleset<NativeBlock>, selector: string, value: Ruleset<NativeBlock>) => {
    ruleset.addNested(selector, value);
  };

  // http://fela.js.org/docs/basics/Rules.html
  handleProperty = (ruleset: Ruleset<NativeBlock>, name: keyof NativeBlock, value: any) => {
    if (name === 'animationName') {
      ruleset.addProperty(name, this.syntax.injectKeyframes(value, this.keyframes).join(', '));
    } else {
      ruleset.addProperty(name, value);
    }
  };

  // http://fela.js.org/docs/basics/Rules.html
  handleSupports = (ruleset: Ruleset<NativeBlock>, query: string, value: Ruleset<NativeBlock>) => {
    ruleset.addNested(`@supports ${query}`, value);
  };

  transformToClassName(styles: (NativeBlock | ParsedBlock)[]): ClassName {
    return this.fela.renderRule(combineRules(...styles.map(style => () => style)), {});
  }
}
