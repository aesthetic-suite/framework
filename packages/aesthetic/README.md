# Aesthetic

[![Build Status](https://travis-ci.org/milesj/aesthetic.svg?branch=master)](https://travis-ci.org/milesj/aesthetic)

Aesthetic is a powerful React library for styling components, whether it be CSS-in-JS using style
objects, importing stylesheets, or simply referencing external class names. Simply put, Aesthetic is
an abstraction layer that utilizes higher-order-components for the compilation of styles via
third-party libraries, all the while providing customizability, theming, and a unified syntax.

```javascript
import React from 'react';
import PropTypes from 'prop-types';
import { StylesPropType } from 'aesthetic';
import withStyles, { classes } from '../path/to/styler';

function Button({ children, styles }) {
  return (
    <button type="button" className={classes(styles.button)}>
      {children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node.isRequire,
  styles: StylesPropType.isRequired,
};

export default withStyles(({ unit }) => ({
  button: {
    textAlign: 'center',
    display: 'inline-block',
    padding: unit,
  },
}))(Button);
```

## Requirements

- React 15/16+
- IE 10+
- `WeakMap`

## Installation

Aesthetic requires React as a peer dependency.

```
npm install aesthetic react --save
// Or
yarn add aesthetic react
```

## Documentation

[https://milesj.gitbook.io/aesthetic](https://milesj.gitbook.io/aesthetic)
