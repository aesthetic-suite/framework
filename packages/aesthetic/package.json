{
  "name": "aesthetic",
  "version": "2.6.0",
  "description": "Aesthetic is a powerful React library for styling components through the use of adapters.",
  "keywords": [
    "aesthetic",
    "react",
    "native",
    "style",
    "components",
    "hoc",
    "css",
    "css-in-js",
    "css-modules",
    "aphrodite",
    "glamor",
    "fela",
    "jss"
  ],
  "repository": "https://github.com/milesj/aesthetic/tree/master/packages/aesthetic",
  "license": "MIT",
  "main": "./lib/index.js",
  "types": "./lib/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "hoist-non-react-statics": "^2.5.5",
    "lodash": "^4.17.10",
    "prop-types": "^15.6.2"
  },
  "peerDependencies": {
    "react": "^15.0.0 || ^16.0.0"
  },
  "devDependencies": {
    "react": "^16.4.1"
  }
}
