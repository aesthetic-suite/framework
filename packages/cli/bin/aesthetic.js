#!/usr/bin/env node

try {
  // Attempt to load it incase some dependencies require it
  // eslint-disable-next-line
  require('regenerator-runtime');
} catch {
  // Ignore if missing
}

require('../lib');
