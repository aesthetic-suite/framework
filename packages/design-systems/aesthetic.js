module.exports = require('./lib/aesthetic/themes/default.js');
