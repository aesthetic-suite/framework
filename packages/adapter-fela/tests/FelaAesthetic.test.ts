/* eslint-disable jest/expect-expect */

import { createRenderer } from 'fela';
// @ts-ignore
import { renderToString } from 'fela-tools';
import webPreset from 'fela-preset-web';
import { Direction } from 'aesthetic';
import {
  cleanStyles,
  convertDirection,
  DIRECTIONS,
  FONT_ROBOTO_FLAT_SRC,
  FONT_CIRCULAR_MULTIPLE_FLAT_SRC,
  KEYFRAME_FADE,
  SYNTAX_UNIFIED_LOCAL_FULL,
  SYNTAX_ATTRIBUTE,
  SYNTAX_DESCENDANT,
  SYNTAX_FALLBACKS,
  SYNTAX_FONT_FACE,
  SYNTAX_FONT_FACE_MIXED,
  SYNTAX_FONT_FACE_MULTIPLE,
  SYNTAX_GLOBAL,
  SYNTAX_KEYFRAMES,
  SYNTAX_MEDIA_QUERY,
  SYNTAX_MULTI_SELECTOR,
  SYNTAX_PROPERTIES,
  SYNTAX_PSEUDO,
  SYNTAX_SUPPORTS,
  FONT_ROBOTO,
  FONT_CIRCULAR_MULTIPLE,
  SYNTAX_MEDIA_QUERY_NESTED,
  SYNTAX_KEYFRAMES_INLINE,
  SYNTAX_FONT_FACES_INLINE,
  SYNTAX_RAW_CSS,
} from 'aesthetic/lib/testUtils';
import FelaAesthetic from '../src/FelaAesthetic';

describe('FelaAesthetic', () => {
  let instance: FelaAesthetic<any>;

  function renderAndTest(
    styleSheet: any,
    expectedStyles: any = {},
    expectedClassName: string = '',
    {
      dir,
      global = false,
    }: {
      dir: Direction;
      global?: boolean;
    },
  ) {
    const options = { name: 'fela', rtl: dir === 'rtl' };
    const convertedSheet = global
      ? instance.syntax.convertGlobalSheet(styleSheet, options).toObject()
      : instance.syntax.convertStyleSheet(styleSheet, options).toObject();
    const parsedSheet = instance.parseStyleSheet(convertedSheet, 'fela');

    expect(convertedSheet).toEqual(convertDirection(expectedStyles, dir));

    expect(instance.transformStyles(Object.values(parsedSheet), options)).toBe(expectedClassName);

    expect(cleanStyles(renderToString(instance.fela))).toMatchSnapshot();
  }

  beforeEach(() => {
    instance = new FelaAesthetic(
      createRenderer({
        plugins: [...webPreset],
      }),
    );
  });

  DIRECTIONS.forEach(dir => {
    // eslint-disable-next-line jest/valid-describe
    describe(dir.toUpperCase(), () => {
      it('converts and transforms inline styles', () => {
        expect(
          instance.transformStyles([{ margin: 0 }, { padding: 2 }], { rtl: dir === 'rtl' }),
        ).toBe('a b');
      });

      describe('global sheet', () => {
        it('handles globals', () => {
          renderAndTest(SYNTAX_GLOBAL, {}, '', { dir, global: true });
        });

        it('handles @font-face', () => {
          const spy = jest.spyOn(instance.fela, 'renderFont');

          instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE, { rtl: dir === 'rtl' });

          expect(spy).toHaveBeenCalledWith('Roboto', FONT_ROBOTO.srcPaths, FONT_ROBOTO_FLAT_SRC);
          expect(spy).toHaveBeenCalledTimes(1);
        });

        it('handles mixed @font-face', () => {
          const spy = jest.spyOn(instance.fela, 'renderFont');

          instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE_MIXED, { rtl: dir === 'rtl' });

          expect(spy).toHaveBeenCalledWith('Roboto', FONT_ROBOTO.srcPaths, FONT_ROBOTO_FLAT_SRC);
          expect(spy).toHaveBeenCalledWith(
            'Circular',
            FONT_CIRCULAR_MULTIPLE[0].srcPaths,
            FONT_CIRCULAR_MULTIPLE_FLAT_SRC[0],
          );
          expect(spy).toHaveBeenCalledTimes(5);
        });

        it('handles multiple @font-face', () => {
          const spy = jest.spyOn(instance.fela, 'renderFont');

          instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE_MULTIPLE, { rtl: dir === 'rtl' });

          expect(spy).toHaveBeenCalledWith(
            'Circular',
            FONT_CIRCULAR_MULTIPLE[0].srcPaths,
            FONT_CIRCULAR_MULTIPLE_FLAT_SRC[0],
          );
          expect(spy).toHaveBeenCalledTimes(4);
        });

        it('handles @keyframes', () => {
          const spy = jest.spyOn(instance.fela, 'renderKeyframe');

          instance.syntax.convertGlobalSheet(SYNTAX_KEYFRAMES, { rtl: dir === 'rtl' });

          expect(spy).toHaveBeenCalledWith(expect.anything(), {});
        });
      });

      describe('style sheet', () => {
        it('converts unified syntax to native syntax and transforms to a class name', () => {
          instance.fela.renderFont('Roboto', FONT_ROBOTO.srcPaths, FONT_ROBOTO);

          instance.keyframes.fade = instance.fela.renderKeyframe(() => KEYFRAME_FADE as any, {});

          renderAndTest(
            SYNTAX_UNIFIED_LOCAL_FULL,
            {
              button: {
                margin: 0,
                padding: '6px 12px',
                border: '1px solid #2e6da4',
                borderRadius: 4,
                display: 'inline-block',
                cursor: 'pointer',
                fontFamily: 'Roboto',
                fontWeight: 'normal',
                lineHeight: 'normal',
                whiteSpace: 'nowrap',
                textDecoration: 'none',
                textAlign: 'left',
                backgroundColor: '#337ab7',
                verticalAlign: 'middle',
                color: 'rgba(0, 0, 0, 0)',
                animationName: 'k1',
                animationDuration: '.3s',
                ':hover': {
                  backgroundColor: '#286090',
                  borderColor: '#204d74',
                },
                '::before': {
                  content: '"â˜…"',
                  display: 'inline-block',
                  verticalAlign: 'middle',
                  marginRight: 5,
                },
                '@media (max-width: 600px)': {
                  padding: '4px 8px',
                },
              },
            },
            'a b c d e f g h i j k l m n o p q r s t u v w x',
            { dir },
          );
        });

        it('handles properties', () => {
          renderAndTest(SYNTAX_PROPERTIES, SYNTAX_PROPERTIES, 'a b c d', { dir });
        });

        it('handles attribute selectors', () => {
          renderAndTest(SYNTAX_ATTRIBUTE, SYNTAX_ATTRIBUTE, 'a b', { dir });
        });

        it('handles descendant selectors', () => {
          renderAndTest(
            SYNTAX_DESCENDANT,
            {
              list: {
                margin: 0,
                padding: 0,
                '> li': {
                  listStyle: 'bullet',
                },
              },
            },
            'a b c',
            { dir },
          );
        });

        it('handles pseudo selectors', () => {
          renderAndTest(SYNTAX_PSEUDO, SYNTAX_PSEUDO, 'a b c', { dir });
        });

        it('handles multiple selectors (comma separated)', () => {
          renderAndTest(
            SYNTAX_MULTI_SELECTOR,
            {
              multi: {
                cursor: 'pointer',
                ':disabled': { cursor: 'default' },
                '[disabled]': { cursor: 'default' },
                '> span': { cursor: 'default' },
              },
            },
            'a b c d',
            { dir },
          );
        });

        it('handles inline @keyframes', () => {
          renderAndTest(
            SYNTAX_KEYFRAMES_INLINE,
            {
              single: {
                animationName: 'k1',
              },
              multiple: {
                animationName: 'k1, unknown, k2',
              },
            },
            'a',
            { dir },
          );
        });

        it('handles inline @font-face', () => {
          renderAndTest(
            SYNTAX_FONT_FACES_INLINE,
            {
              single: {
                fontFamily: 'Roboto',
              },
              multiple: {
                fontFamily: 'Circular, OtherFont, Roboto',
              },
            },
            'a',
            { dir },
          );
        });

        it('handles @fallbacks', () => {
          renderAndTest(
            SYNTAX_FALLBACKS,
            {
              fallback: {
                background: ['red', 'linear-gradient(...)'],
                display: ['block', 'inline-block', 'flex'],
                color: ['blue'],
              },
            },
            'a b c',
            { dir },
          );
        });

        it('handles @media', () => {
          renderAndTest(
            SYNTAX_MEDIA_QUERY,
            {
              media: {
                color: 'red',
                paddingLeft: 10,
                '@media (max-width: 1000px)': {
                  color: 'green',
                  paddingLeft: 20,
                },
                '@media (min-width: 300px)': {
                  color: 'blue',
                  paddingLeft: 15,
                },
              },
            },
            'a b c d e f',
            { dir },
          );
        });

        it('handles nested @media', () => {
          renderAndTest(
            SYNTAX_MEDIA_QUERY_NESTED,
            {
              media: {
                color: 'red',
                '@media (min-width: 300px)': {
                  color: 'blue',
                  '@media (max-width: 1000px)': {
                    color: 'green',
                  },
                },
              },
            },
            'a b c',
            { dir },
          );
        });

        it('handles @supports', () => {
          renderAndTest(
            SYNTAX_SUPPORTS,
            {
              sup: {
                display: 'block',
                '@supports (display: flex)': {
                  display: 'flex',
                },
                '@supports not (display: flex)': {
                  float: 'left',
                },
              },
            },
            'a b c',
            { dir },
          );
        });

        it('handles raw CSS', () => {
          renderAndTest(SYNTAX_RAW_CSS, {}, '', { dir });
        });
      });
    });
  });
});
