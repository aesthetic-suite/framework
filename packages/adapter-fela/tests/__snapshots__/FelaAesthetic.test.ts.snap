// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FelaAesthetic LTR global sheet handles globals 1`] = `"body{margin:0}html{height:100%}a{color:red}"`;

exports[`FelaAesthetic LTR style sheet converts unified syntax to native syntax and transforms to a class name 1`] = `"@font-face{font-family:\\"Roboto\\";font-style:normal;font-weight:normal;src:url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype')}@-webkit-keyframes k1{from{opacity:0}to{opacity:1}}@-moz-keyframes k1{from{opacity:0}to{opacity:1}}@keyframes k1{from{opacity:0}to{opacity:1}}.a{margin:0}.b{padding:6px 12px}.c{border:1px solid #2e6da4}.d{border-radius:4px}.e{display:inline-block}.f{cursor:pointer}.g{font-family:Roboto}.h{font-weight:normal}.i{line-height:normal}.j{white-space:nowrap}.k{text-decoration:none}.l{text-align:left}.m{background-color:#337ab7}.n{vertical-align:middle}.o{color:rgba(0, 0, 0, 0)}.p{animation-name:k1}.q{animation-duration:.3s}.t::before{content:\\"★\\"}.u::before{display:inline-block}.v::before{vertical-align:middle}.w::before{margin-right:5px}.r:hover{background-color:#286090}.s:hover{border-color:#204d74}@media (max-width: 600px){.x{padding:4px 8px}}"`;

exports[`FelaAesthetic LTR style sheet handles @fallbacks 1`] = `".a{background:red;background:-webkit-linear-gradient(...);background:-moz-linear-gradient(...);background:linear-gradient(...)}.b{display:block;display:inline-block;display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}.c{color:blue}"`;

exports[`FelaAesthetic LTR style sheet handles @media 1`] = `".a{color:red}.b{padding-left:10px}@media (min-width: 300px){.c{color:blue}.d{padding-left:15px}}@media (max-width: 1000px){.e{color:green}.f{padding-left:20px}}"`;

exports[`FelaAesthetic LTR style sheet handles @supports 1`] = `".a{display:block}@supports (display: flex){.b{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}}@supports not (display: flex){.c{float:left}}"`;

exports[`FelaAesthetic LTR style sheet handles attribute selectors 1`] = `".a{display:block}.b[disabled]{opacity:0.5}"`;

exports[`FelaAesthetic LTR style sheet handles descendant selectors 1`] = `".a{margin:0}.b{padding:0}.c> li{list-style:bullet}"`;

exports[`FelaAesthetic LTR style sheet handles inline @font-face 1`] = `"@font-face{font-family:\\"Roboto\\";font-style:normal;font-weight:normal;src:url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:normal;src:url('fonts/Circular.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:italic;font-weight:normal;src:url('fonts/Circular-Italic.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:300;src:url('fonts/Circular-Light.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:700;src:url('fonts/Circular-Bold.woff2') format('woff2')}.a{font-family:Circular, OtherFont, Roboto}"`;

exports[`FelaAesthetic LTR style sheet handles inline @keyframes 1`] = `"@-webkit-keyframes k1{0%{left:0%}50%{left:50%}100%{left:100%}}@-moz-keyframes k1{0%{left:0%}50%{left:50%}100%{left:100%}}@keyframes k1{0%{left:0%}50%{left:50%}100%{left:100%}}@-webkit-keyframes k2{from{opacity:0}to{opacity:1}}@-moz-keyframes k2{from{opacity:0}to{opacity:1}}@keyframes k2{from{opacity:0}to{opacity:1}}.a{animation-name:k1, unknown, k2}"`;

exports[`FelaAesthetic LTR style sheet handles multiple selectors (comma separated) 1`] = `".a{cursor:pointer}.b:disabled{cursor:default}.c[disabled]{cursor:default}.d> span{cursor:default}"`;

exports[`FelaAesthetic LTR style sheet handles nested @media 1`] = `".a{color:red}@media (min-width: 300px){.b{color:blue}}@media (min-width: 300px) and (max-width: 1000px){.c{color:green}}"`;

exports[`FelaAesthetic LTR style sheet handles properties 1`] = `".a{color:black}.b{display:inline}.c{margin-right:10px}.d{padding:0}"`;

exports[`FelaAesthetic LTR style sheet handles pseudo selectors 1`] = `".a{position:fixed}.c::before{position:absolute}.b:hover{position:static}"`;

exports[`FelaAesthetic LTR style sheet handles raw CSS 1`] = `".fela-button{display:'block';font-size:16px;text-align:'left';}.fela-button:hover{color:'red';}.fela-button{vertical-align:'middle';}@media (max-width:600px){.fela-button{display:'none';}}@supports (display:-webkit-box) or (display:-webkit-flex) or (display:-ms-flexbox) or (display:flex){.fela-button{display:'flex';}}"`;

exports[`FelaAesthetic RTL global sheet handles globals 1`] = `"body{margin:0}html{height:100%}a{color:red}"`;

exports[`FelaAesthetic RTL style sheet converts unified syntax to native syntax and transforms to a class name 1`] = `"@font-face{font-family:\\"Roboto\\";font-style:normal;font-weight:normal;src:url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype')}@-webkit-keyframes k1{from{opacity:0}to{opacity:1}}@-moz-keyframes k1{from{opacity:0}to{opacity:1}}@keyframes k1{from{opacity:0}to{opacity:1}}.a{margin:0}.b{padding:6px 12px}.c{border:1px solid #2e6da4}.d{border-radius:4px}.e{display:inline-block}.f{cursor:pointer}.g{font-family:Roboto}.h{font-weight:normal}.i{line-height:normal}.j{white-space:nowrap}.k{text-decoration:none}.l{text-align:right}.m{background-color:#337ab7}.n{vertical-align:middle}.o{color:rgba(0, 0, 0, 0)}.p{animation-name:k1}.q{animation-duration:.3s}.t::before{content:\\"★\\"}.u::before{display:inline-block}.v::before{vertical-align:middle}.w::before{margin-left:5px}.r:hover{background-color:#286090}.s:hover{border-color:#204d74}@media (max-width: 600px){.x{padding:4px 8px}}"`;

exports[`FelaAesthetic RTL style sheet handles @fallbacks 1`] = `".a{background:red;background:-webkit-linear-gradient(...);background:-moz-linear-gradient(...);background:linear-gradient(...)}.b{display:block;display:inline-block;display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}.c{color:blue}"`;

exports[`FelaAesthetic RTL style sheet handles @media 1`] = `".a{color:red}.b{padding-right:10px}@media (min-width: 300px){.c{color:blue}.d{padding-right:15px}}@media (max-width: 1000px){.e{color:green}.f{padding-right:20px}}"`;

exports[`FelaAesthetic RTL style sheet handles @supports 1`] = `".a{display:block}@supports (display: flex){.b{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}}@supports not (display: flex){.c{float:right}}"`;

exports[`FelaAesthetic RTL style sheet handles attribute selectors 1`] = `".a{display:block}.b[disabled]{opacity:0.5}"`;

exports[`FelaAesthetic RTL style sheet handles descendant selectors 1`] = `".a{margin:0}.b{padding:0}.c> li{list-style:bullet}"`;

exports[`FelaAesthetic RTL style sheet handles inline @font-face 1`] = `"@font-face{font-family:\\"Roboto\\";font-style:normal;font-weight:normal;src:url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:normal;src:url('fonts/Circular.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:italic;font-weight:normal;src:url('fonts/Circular-Italic.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:300;src:url('fonts/Circular-Light.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:700;src:url('fonts/Circular-Bold.woff2') format('woff2')}.a{font-family:Circular, OtherFont, Roboto}"`;

exports[`FelaAesthetic RTL style sheet handles inline @keyframes 1`] = `"@-webkit-keyframes k1{0%{right:0%}50%{right:50%}100%{right:100%}}@-moz-keyframes k1{0%{right:0%}50%{right:50%}100%{right:100%}}@keyframes k1{0%{right:0%}50%{right:50%}100%{right:100%}}@-webkit-keyframes k2{from{opacity:0}to{opacity:1}}@-moz-keyframes k2{from{opacity:0}to{opacity:1}}@keyframes k2{from{opacity:0}to{opacity:1}}.a{animation-name:k1, unknown, k2}"`;

exports[`FelaAesthetic RTL style sheet handles multiple selectors (comma separated) 1`] = `".a{cursor:pointer}.b:disabled{cursor:default}.c[disabled]{cursor:default}.d> span{cursor:default}"`;

exports[`FelaAesthetic RTL style sheet handles nested @media 1`] = `".a{color:red}@media (min-width: 300px){.b{color:blue}}@media (min-width: 300px) and (max-width: 1000px){.c{color:green}}"`;

exports[`FelaAesthetic RTL style sheet handles properties 1`] = `".a{color:black}.b{display:inline}.c{margin-left:10px}.d{padding:0}"`;

exports[`FelaAesthetic RTL style sheet handles pseudo selectors 1`] = `".a{position:fixed}.c::before{position:absolute}.b:hover{position:static}"`;

exports[`FelaAesthetic RTL style sheet handles raw CSS 1`] = `".fela-button{display:'block';font-size:16px;text-align:'right';}.fela-button:hover{color:'red';}.fela-button{vertical-align:'middle';}@media (max-width:600px){.fela-button{display:'none';}}@supports (display:-webkit-box) or (display:-webkit-flex) or (display:-ms-flexbox) or (display:flex){.fela-button{display:'flex';}}"`;
